hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -508,6 +515,19 @@ private boolean isZombie(K id) {
 		return prop -> prop.getter().apply(defaultEntity);
 	}
 
+	private List<MerkleLeafChanges<K, A, P>> getAccumulatedMerkleLeafChanges() {
+		final List<MerkleLeafChanges<K, A, P>> merkleLeavesChanges = new ArrayList<>();
```

Comment:
Instead of re-creating this `ArrayList` every call to `commit()`, imo we should follow the same pattern used with e.g. the `List<K> createdKeys`---that is, have a single instance that we `clear()` in `begin()`. 

The cost of creating a new `ArrayList` every handle is actually noticeable in profiling.

Reply:
That's a great idea and I applied it! However, do you think we should clear this merkleLeavesChanges collection in `begin()`? Wouldn't it be an overdesign, since we also clear this collection in `commit()`?